packages/compiler-core/src/transforms/hoistStatic.ts

```diff
@@ -116,23 +116,24 @@ function isStaticNode(
       if (node.tagType !== ElementTypes.ELEMENT) {
         return false
       }
-      if (resultCache.has(node)) {
-        return resultCache.get(node) as boolean
-      }
-      const flag = getPatchFlag(node)
-      if (!flag || flag === PatchFlags.TEXT) {
-        // element self is static. check its children.
-        for (let i = 0; i < node.children.length; i++) {
-          if (!isStaticNode(node.children[i], resultCache)) {
-            resultCache.set(node, false)
-            return false
+      const cached = resultCache.get(node)
+      if (cached === undefined) {
```

Comment:
you can't, needs to be `cached !== undefined` if you want to return early, since the `cached` can be either `true` or `false`

I think `cached === undefined` reads better and is safer than `cached !== undefined`.

I don't feel too strong either way tbh, can change to `if(cached !== undefined)` if needed :)